test/gtest/pad_constant.hpp

```diff
@@ -0,0 +1,218 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include "get_handle.hpp"
+#include "miopen/allocator.hpp"
+#include "random.hpp"
+#include "verify.hpp"
+#include "tensor_holder.hpp"
+#include "cpu_pad_constant.hpp"
+#include <cstdio>
+#include <gtest/gtest.h>
+#include <miopen/pad_constant.hpp>
+
+struct PadConstantTestCase
+{
+    size_t N;
+    size_t C;
+    size_t D;
+    size_t H;
+    size_t W;
+
+    friend std::ostream& operator<<(std::ostream& os, const PadConstantTestCase& tc)
+    {
+        return os << "(N: " << tc.N << " C:" << tc.C << " D:" << tc.D << " H:" << tc.H
+                  << " W:" << tc.W << ")";
+    }
+
+    std::vector<size_t> GetInput()
```

Comment:
Is there a reason you put 1 where there is no value?

Reply:
I think I did that because the CPU portion of the tests didn't have the consistency checks, so I would use `1` instead of just omitting it

Reply:
moDNN's constant padding supports both 1D and 5D. Test all of these cases to see if they work.

Reply:
I meant the `cpu_pad_constant` portion to generate the reference output within gtest, the kernel works properly as expected with less dimensions supplied (I tested it with MIOpenDriver). I'll fix it